### PR TITLE
CI-ARTIFACTS update NFD to run nightly master

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
Signed-off-by: Carlos Eduardo Arango Gutierrez <carangog@redhat.com>

### CODE REVIEW

File: ci-operator/config/openshift-psap/ci-artifacts/openshift-psap-ci-artifacts-master__nfd.yaml [hunk 1]
```diff
@@ -21,13 +21,14 @@ resources:
       memory: 2Gi
 tests:
 - as: operator-e2e-master
+  cron: 0 23 * * *
```
Comment (kpouget) [inline ci-operator/config/openshift-psap/ci-artifacts/openshift-psap-ci-artifacts-master__nfd.yaml:24]: wrong place :) must be below the cluster_claim

Comment (ArangoGutierrez) [reply ci-operator/config/openshift-psap/ci-artifacts/openshift-psap-ci-artifacts-master__nfd.yaml]: Fixed

File: ci-operator/jobs/openshift-psap/ci-artifacts/openshift-psap-ci-artifacts-master-presubmits.yaml [hunk 2]
```diff
@@ -1057,7 +1057,7 @@ presubmits:
           secretName: result-aggregator
     trigger: (?m)^/test( | .* )nfd-images,?($|\s.*)
   - agent: kubernetes
-    always_run: false
+    always_run: true
```
Comment (kpouget) [inline ci-operator/jobs/openshift-psap/ci-artifacts/openshift-psap-ci-artifacts-master-presubmits.yaml:1060]: I don't think we need `always_run: true` on `ci-artifacts` PR, rigth?
(there's a way to "always run if some files are modified, Ecosystem team did it in their fork, but I didn't try it yet)

Comment (ArangoGutierrez) [reply ci-operator/jobs/openshift-psap/ci-artifacts/openshift-psap-ci-artifacts-master-presubmits.yaml]: done

Comment (sagidayan) [reply ci-operator/jobs/openshift-psap/ci-artifacts/openshift-psap-ci-artifacts-master-presubmits.yaml]: You can take a look here https://github.com/openshift/release/blob/2849dfe791a13176875264a23928397b644204ab/ci-operator/config/rh-ecosystem-edge/ci-artifacts/rh-ecosystem-edge-ci-artifacts-master.yaml#L34

File: ci-operator/config/openshift-psap/ci-artifacts/openshift-psap-ci-artifacts-master__nfd.yaml [hunk 3]
```diff
@@ -38,6 +38,25 @@ tests:
...
+  steps:
+    test:
+    - as: nightly
+      commands: run prow nfd-operator test_master_branch
+      from: ci-artifacts
+      resources:
+        requests:
+          cpu: 2000m
+          memory: 2Gi
+    workflow: ipi-aws
```
Comment (kpouget) [inline ci-operator/config/openshift-psap/ci-artifacts/openshift-psap-ci-artifacts-master__nfd.yaml:59]:
```
    workflow: generic-claim
```

Comment (kpouget) [reply ci-operator/config/openshift-psap/ci-artifacts/openshift-psap-ci-artifacts-master__nfd.yaml:59]: when you take the cluster from a pool (`cluster_claim`), you need `workflow: generic-claim`

Comment (kpouget) [reply ci-operator/config/openshift-psap/ci-artifacts/openshift-psap-ci-artifacts-master__nfd.yaml:59]: while `ipi-aws` is when you want it built

Comment (kpouget) [reply ci-operator/config/openshift-psap/ci-artifacts/openshift-psap-ci-artifacts-master__nfd.yaml:59]: ... but in Periodics, you can't use the pool IIRC

**Comment by ArangoGutierrez** [comment]
RFR @kpouget

**Comment by kpouget** [comment]
seems good to me

/lgtm
/hold | merge when you want

**Pull request merged**